src/components/Card.js

```diff
@@ -1,52 +1,63 @@
 export default class Card {
-    constructor(data, cardSelector, handleImageClick) {
+  constructor(data, cardSelector, handleImageClick, handleCardDelete, handleCardLike) {
     this._data = data;
     this._name = data.name;
     this._link = data.link;
-    this._cardSelector = cardSelector; 
+    this.cardSelector = cardSelector;
     this._element = this._getTemplate();
     this._handleImageClick = handleImageClick;
+    this._handleCardDelete = handleCardDelete;
     this._cardLikeBtn = this._element.querySelector(".card__like-button");
+    this._isLiked = data.like
   }
 
   _getTemplate() {
- const cardTemplate = document.querySelector(this._cardSelector).content.querySelector(".card").cloneNode(true);
- return cardTemplate;
+    const cardTemplate = document
+      .querySelector(this.cardSelector)
+      .content.querySelector(".card")
+      .cloneNode(true);
+    return cardTemplate;
   }
-  
+
   _setEventListeners() {
- 
-    this._cardLikeBtn.addEventListener('click', () =>
-        this._handleCardLike());
+    this._cardLikeBtn.addEventListener("click", () => this._handleCardLike());
 
-      this._element.querySelector("#card-delete-btn").addEventListener('click', () =>
-        this._handleCardDelete());
+    this._element
+      .querySelector("#card-trash-btn")
+      .addEventListener("click", () => {
+        this._handleCardDelete(this);
+      });
 
-      // pass data to the image click handler
-      this._element.querySelector('.card__image').addEventListener('click', () => 
-         this._handleImageClick(this._data));
+    // pass data to the image click handler
+    this._element
+      .querySelector(".card__image")
+      .addEventListener("click", () => this._handleImageClick(this._data));
   }
 
   _handleCardLike() {
-    this._cardLikeBtn.classList.toggle('card__like-button_active');
-  
+    this._cardLikeBtn.classList.toggle("card__like-button_active");
   }
 
-  _handleCardDelete() {
-      this._element.remove();
-      this._element = null;
+  deleteCard() {
+    this._element.remove();
+    this._element = null;
   }
 
   cardView() {
-   // get the card view
-   // set event listeners
-   // return the card
-   
-   this._element.querySelector('.card__header').textContent = this._name;
-   this._element.querySelector('.card__image').src = this._link;
-   this._element.querySelector('.card__image').alt = this._name;
-
-   this._setEventListeners();
-   return this._element;
+    // get the card view
+    // set event listeners
+    // return the card
+
+    this._element.querySelector(".card__header").textContent = this._name;
+    this._element.querySelector(".card__image").src = this._link;
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```